### PR TITLE
OS-8584 Introduce gcc14 as the shadow-compiler

### PR DESCRIPTION
**Pull request by tsoome**
Update Jenkinsfile for gcc14

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -255,7 +255,7 @@ export PLAT_CONFIGURE_ARGS="-d $PLAT_CONFIGURE_ARGS"
                     node {
```
Comment (danmcd) [inline Jenkinsfile:255]: A couple lines above, you need to rename the stage to `gcc14` as well (line 253).

**Review by danmcd — CHANGES_REQUESTED** [review]
Update copyright, and rename stage.  There should be 11 changed lines of gcc7 -> gcc14, plus a copyright update.

File: Jenkinsfile (hunk 1 above)
Comment (tsoome) [reply Jenkinsfile:255]: fixed.

**Review by danmcd — APPROVED** [review]
Will grant IA if the Jenkins build works!

**Pull request merged**